sigstore-java/src/main/java/dev/sigstore/timestamp/client/TimestampClientHttp.java

```diff
@@ -0,0 +1,167 @@
+/*
+ * Copyright 2025 The Sigstore Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package dev.sigstore.timestamp.client;
+
+import com.google.api.client.http.ByteArrayContent;
+import com.google.api.client.http.GenericUrl;
+import com.google.api.client.http.HttpRequestFactory;
+import com.google.api.client.http.HttpResponse;
+import com.google.common.annotations.VisibleForTesting;
+import dev.sigstore.http.HttpClients;
+import dev.sigstore.http.HttpParams;
+import dev.sigstore.http.ImmutableHttpParams;
+import java.io.IOException;
+import java.net.URI;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Objects;
+import org.bouncycastle.asn1.ASN1ObjectIdentifier;
+import org.bouncycastle.tsp.TSPAlgorithms;
+import org.bouncycastle.tsp.TSPException;
+import org.bouncycastle.tsp.TimeStampRequest;
+import org.bouncycastle.tsp.TimeStampRequestGenerator;
+import org.bouncycastle.tsp.TimeStampResponse;
+
+/** A client to communicate with a timestamp service instance. */
+public class TimestampClientHttp implements TimestampClient {
+  private static final URI SIGSTORE_TSA_URI =
+      URI.create("https://timestamp.sigstage.dev/api/v1/timestamp");
+  private static final String CONTENT_TYPE_TIMESTAMP_QUERY = "application/timestamp-query";
+  private static final String ACCEPT_TYPE_TIMESTAMP_REPLY = "application/timestamp-reply";
+
+  private static final Map<String, ASN1ObjectIdentifier> HASH_ALGORITHM_OIDS;
+
+  static {
```

Comment:
this can be created on one line kinda like:

```
    private static final ImmutableMap<String, ASN1ObjectIdentifier> HASH_ALGORITHM_OIDS =
            ImmutableMap.of(
                    "SHA256", TSPAlgorithms.SHA256,
                    "SHA384", TSPAlgorithms.SHA384,
                    "SHA512", TSPAlgorithms.SHA512
            );
```

Reply:
alternatively, we could add an Enum to TimestampRequest that strictly defines these supported hashes, and we wouldn't have to check them later.